src/features/comments.ts

```diff
@@ -0,0 +1,64 @@
+/* eslint-disable no-param-reassign */
+
+import { createAsyncThunk, createSlice } from '@reduxjs/toolkit';
+import { Comment } from '../types/Comment';
+import { createComment, deleteComment, getPostComments } from '../api/comments';
+
+export const getComments = createAsyncThunk(
+  'comments/fetch',
+  (postId: number) => getPostComments(postId),
+);
+
+export const createNewComment = createAsyncThunk(
+  'comments/create',
+  (data: Omit<Comment, 'id'>) => createComment(data),
+);
+
+export const deleteAPIComment = createAsyncThunk(
+  'comments/delete',
+  (commentId: number) => deleteComment(commentId),
+);
+
+type Users = {
+  comments: Comment[];
+  loaded: boolean;
+  hasError: boolean;
+};
+const initialState: Users = {
+  comments: [],
+  loaded: true,
+  hasError: false,
+};
+
+const commentsSlice = createSlice({
+  name: 'comments',
+  initialState,
+  reducers: {
+    create: () => {},
+    delete: (state, action) => {
+      deleteComment(action.payload);
```

Comment:
The `delete` reducer directly calls `deleteComment`, which is an API call. This is not recommended as reducers should be pure functions. Consider handling this logic in a thunk or an extra reducer.